ignite/contrib/handlers/tqdm_logger.py

```
@@ -0,0 +1,115 @@
+        the correct format of the stdout.
+    """
+
+    def __init__(self):
+        self.pbar = None
+
+    def _reset(self, engine):
+        self.pbar = tqdm(
+            total=len(engine.state.dataloader),
+            leave=False,
+            bar_format='{desc}[{n_fmt}/{total_fmt}] {percentage:3.0f}%|{bar}{postfix} [{elapsed}<{remaining}]')
+
+    def _close(self, engine):
+        self.pbar.close()
+
+    def _update(self, engine, metric_names=None):
+        self.pbar.set_description('Epoch {}'.format(engine.state.epoch))
+
+        if metric_names is not None:
+            metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+            self.pbar.set_postfix(**metrics)
+
+        self.pbar.update()
+
+    @staticmethod
+    def log_message(message):
+        """
+        Logs a message, preserving the progress bar correct output format
+
+        Args:
+            message (str): string you wish to log
+        """
+        tqdm.write(message)
+
+    def _log_metrics(self, engine, metric_names, mode='epoch', log_interval=1):
+
+        i = engine.state.epoch if mode == 'epoch' else engine.state.iteration
+
+        if log_interval and i % log_interval == 0:
+            if mode == 'epoch':
+                message = 'Epoch {}'.format(engine.state.epoch)
+            else:
+                message = 'Iteration {}'.format(engine.state.iteration)
+
+            metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+            for name, value in metrics.items():
+                message += ' | {}={}'.format(name, value)
+
+            self.log_message(message)
+
+    def attach(self, engine, metric_names=None):
+        """
+        Attaches the progress bar to an engine object
+
+        Args:
+            engine (Engine): engine object
+            metric_names (list): (Optional) list of the metrics names to log as the bar progresses
+        """
+
```

please add a check on the type `list` here. This could avoid silent errors like `pbar.attach(trainer, "avg_accuracy")`